src/blockdata/transaction.rs

```diff
@@ -503,6 +503,7 @@ impl<E> EncodeSigningDataResult<E> {
     ///     // use a hash value of "1", instead of computing the actual hash due to SIGHASH_SINGLE bug
     /// }
     /// ```
+    #[allow(clippy::wrong_self_convention)] // E is not Copy so we consume self.
     pub fn is_sighash_single_bug(self) -> Result<bool, E> {
```

Comment:
I knew this was ugly when I saw it the first time. Sadly, I have no idea how to improve the name.

Reply:
Ha that's funny, I thought the same myself on both accounts.